mlflow/utils/proto_json_utils.py

```diff
@@ -97,13 +95,26 @@ def _dataframe_from_json(path_or_str, schema: Schema = None,
     :return: pandas.DataFrame.
     """
     if schema is not None:
-        dtypes = dict(zip(schema.column_names(), schema.column_types()))
-        df = pd.read_json(path_or_str, orient=pandas_orient, dtype=dtypes)[schema.column_names()]
+        dtypes = dict(zip(schema.column_names(), schema.numpy_types()))
+        df = pd.read_json(path_or_str, orient=pandas_orient, dtype=dtypes)
+
+        names = schema.column_names()
+        expected_names = set(names)
+        actual_names = set(df.columns)
+        if actual_names != expected_names:
+            missing_cols = expected_names - actual_names
+            extra_columns = actual_names - expected_names
+            raise MlflowException("Incompatible data. {0} {1}".format(
+                "{}.".format("Missing columns {}".format(missing_cols) if missing_cols else ""),
```

Comment:
```suggestion
                "Missing columns {}".format(missing_cols) if missing_cols else "",
```

Nit: remove redundant call to `.format`? Same with the next line

Reply:
It's not redundant.